src/main/java/de/dennisguse/opentracks/io/file/exporter/ExportActivity.java

```diff
@@ -91,7 +92,7 @@
     private ContentProviderUtils contentProviderUtils;
 
     // List of tracks to be exported.
-    private final ArrayList<Track.Id> trackIds = new ArrayList<>();
+    private final ArrayList<UUID> uuids = new ArrayList<>();
```

Comment:
That is the only place where we are passing the UUID around.
What is the advantage over passing the Track.Id or the Track?

Reply:
Because export method in `ExportActivity` (line 236) needs UUID to check if file already exists. Before it used Track.Id but now it needs UUID.

Also, PendingConflict calls that export method and the only thing it knows is UUID (before Track.Id).

PS/ I thought the same is to use Track.Id than to use UUID to identify tracks. I you are not sure about it I could re-think it.

Reply:
It will work; it would just be the first time we use it internally.
What about using ArrayList<Track> instead?
The track is loaded anyhow to get the UUID, right?

Reply:
Just realized that `<Track>` was hidden as I did not put it into backticks.